src/simpleweightedgraph.jl

```diff
@@ -18,7 +18,7 @@ mutable struct SimpleWeightedGraph{T<:Integer, U<:Real} <: AbstractSimpleWeighte
 
 end
 
-ne(g::SimpleWeightedGraph) = nnz(g.weights) ÷ 2
+ne(g::SimpleWeightedGraph) = (nnz(g.weights) + count(diag(g.weights) .!= 0)) ÷ 2
```

Comment:
I think the assumption is, that `ne` is kind of fast. Ideally we would solve this by either storing the number of edges or the number of self-loops in a separate field, but I guess for now we can also just query for the diagonal elements.

But `diag` does also allocate a new Vector, so I think we should at least some non-allocating function here.

Reply:
Maybe `nselfloop` helps?